packages/frontend/src/hooks/contracts/useTokenBalance.ts

```diff
@@ -8,15 +8,15 @@ import { useWallet } from '@context/wallet'
 import { toTokenAmount } from '@utils/calculations'
 
 const tokenBalanceQueryKeys = {
-  userTokenBalance: () => ['userTokenBalance'],
+  userTokenBalance: (token: string) => ['userTokenBalance', token],
```

Comment:
Having token (address) as a key is critical because the crab loading state was mixed with another loading state of getting the token.